Document/0x06c-Reverse-Engineering-and-Tampering.md

```diff
@@ -347,6 +347,118 @@ zsh: # . ~/.zshrc
+
+Using `lsof` list of various open files for an iOS application running with PID 28282 is shown below.
+
+```
+iPhone:~ root# lsof -p 2828
+COMMAND  PID   USER   FD   TYPE DEVICE SIZE/OFF   NODE NAME
+iOweApp 2828 mobile  cwd    DIR    1,2      864      2 /
+iOweApp 2828 mobile  txt    REG    1,3   206144 189774 /private/var/containers/Bundle/Application/F390A491-3524-40EA-B3F8-6C1FA105A23A/iOweApp.app/iOweApp
+iOweApp 2828 mobile  txt    REG    1,3     5492 213230 /private/var/mobile/Containers/Data/Application/5AB3E437-9E2D-4F04-BD2B-972F6055699E/tmp/com.apple.dyld/iOweApp-6346DC276FE6865055F1194368EC73CC72E4C5224537F7F23DF19314CF6FD8AA.closure
+iOweApp 2828 mobile  txt    REG    1,3    30628 212198 /private/var/preferences/Logging/.plist-cache.vqXhr1EE
+iOweApp 2828 mobile  txt    REG    1,2    50080 234433 /usr/lib/libobjc-trampolines.dylib
+iOweApp 2828 mobile  txt    REG    1,2   344204  74185 /System/Library/Fonts/AppFonts/ChalkboardSE.ttc
+iOweApp 2828 mobile  txt    REG    1,2   664848 234595 /usr/lib/dyld
+iOweApp 2828 mobile  txt    REG    1,2 30183936 235695 /usr/share/icu/icudt64l.dat
+iOweApp 2828 mobile  txt    REG    1,2 92936040  74056 /System/Library/CoreServices/CoreGlyphs.bundle/Assets.car
+iOweApp 2828 mobile  txt    REG    1,2  2169192  75023 /System/Library/Fonts/CoreUI/SFUI.ttf
+iOweApp 2828 mobile  txt    REG    1,2  1202584  75408 /System/Library/Fonts/Core/Helvetica.ttc
+iOweApp 2828 mobile    0r   CHR    3,2      0t0    197 /dev/null
+iOweApp 2828 mobile    1u   CHR    3,2      0t0    197 /dev/null
+iOweApp 2828 mobile    2u   CHR    3,2    0t141    197 /dev/null
+```
+
+##### Opened Connections
+
+`lsof` command when invoved with option `-i`, it gives the list of open network ports for all active processes on the device. To get a list of open network ports for a specific process, the `lsof -i -a -p <pid>` command can be used, where `-a` (AND) option is used for filtering. Below a filtered output for PID 1 is shown.
+
+```
+iPhone:~ root# lsof -i -a -p 1
+COMMAND PID USER   FD   TYPE             DEVICE SIZE/OFF NODE NAME
+launchd   1 root    2u  IPv6 0x69c2ce210efdba03      0t0  TCP localhost:mpm-flags (LISTEN)
+launchd   1 root    3u  IPv4 0x69c2ce21125bc873      0t0  TCP localhost:mpm-flags (LISTEN)
+launchd   1 root    8u  IPv6 0x69c2ce210efdcc63      0t0  TCP localhost:intu-ec-client (LISTEN)
+launchd   1 root    9u  IPv4 0x69c2ce210eeac873      0t0  TCP localhost:intu-ec-client (LISTEN)
+launchd   1 root   12u  IPv4 0x69c2ce210eeabebb      0t0  TCP localhost:socks (LISTEN)
+launchd   1 root   13u  IPv4 0x69c2ce210eeadbe3      0t0  TCP localhost:ansoft-lm-1 (LISTEN)
+launchd   1 root   14u  IPv6 0x69c2ce210efdc643      0t0  TCP *:62078 (LISTEN)
```

Comment:
Could you please only list the first 4-5 lines and then make "..." and remove the rest?